lib/lib-storage/src/Upload.ts

```diff
@@ -212,7 +213,25 @@ export class Upload extends EventEmitter {
       };
       result = await this.client.send(new CompleteMultipartUploadCommand(uploadCompleteParams));
     } else {
-      result = this.putResponse!;
+      const endpoint = await this.client.config.endpoint();
+      const locationKey = this.params
+        .Key!.split("/")
+        .map((segment) => extendedEncodeURIComponent(segment))
+        .join("/");
+      const locationBucket = extendedEncodeURIComponent(this.params.Bucket!);
+
+      let Location: string;
+      if (this.client.config.forcePathStyle) {
+        Location = `${endpoint.protocol}//${endpoint.hostname}/${locationBucket}/${locationKey}`;
+      } else {
+        Location = `${endpoint.protocol}//${locationBucket}.${endpoint.hostname}/${locationKey}`;
+      }
+      result = {
+        ...this.putResponse!,
```

Comment:
Wouldn't it make more sense to add it directly to the object (although you would probably need to change some types)?